OpenRA.Mods.Common/Warheads/TargetDamageWarhead.cs

```diff
@@ -35,20 +34,30 @@ protected override void DoImpact(WPos pos, Actor firedBy, WarheadArgs args)
 				if (!IsValidAgainst(victim, firedBy))
 					continue;
 
-				var closestActiveShape = victim.TraitsImplementing<HitShape>()
-					.Where(Exts.IsTraitEnabled)
-					.Select(s => (HitShape: s, Distance: s.DistanceFromEdge(victim, pos)))
-					.MinByOrDefault(s => s.Distance);
+				HitShape closestActiveShape = null;
+				var closestDistance = new WDist(int.MaxValue);
+				foreach (var targetPos in victim.GetEnabledTargetablePositions())
+				{
+					if (targetPos is HitShape h)
+					{
+						var distance = h.DistanceFromEdge(victim, pos);
```

Comment:
wouldn't it be better to use ints? The we wouldn't need to use `new WDist(int.MaxValue)`
```suggestion
						var distance = h.DistanceFromEdge(victim, pos).Length;
```